collector/src/main/java/com/bitmovin/analytics/stateMachines/PlayerStateMachine.java

```diff
@@ -173,6 +182,32 @@ public void setVideoStartFailedReason(VideoStartFailedReason videoStartFailedRea
         this.videoStartFailedReason = videoStartFailedReason;
     }
 
+    public CountDownTimer videoStartTimeout = new CountDownTimer(Util.VIDEOSTART_TIMEOUT, 1000) {
+        @Override
+        public void onTick(long millisUntilFinished) {
```

Comment:
Does it maybe make sense to make this method protected and move entire timer logic to the state machine? AFAICS already everything is happening inside `PlayerState`, so only change would be to move stopping timer `onError`?

Reply:
Thanks for this nice catch 👍 
https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/113/commits/cbc59fbeb7730c124364265743594df9cb513459